File: evanescent/evan.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = input()
        ans = 1
        x = 0
        for i in range(1, n):
            if a[i] != a[i - 1]:
                ans += 1
            if i == n - 1:
                break
            if a[i] != a[i + 1] and a[i] != a[i - 1]:
                if a[i - 1] == a[i + 1]:
                    x = 2
                else:
                    x = max(x, 1)
        print(ans - x)
    
    
if __name__ == "__main__":
    main()

File: evanescent/evan.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int t;
    cin >> t;
    while(t--) {
        int n;
        cin >> n;
        string s;
        cin >> s;
        
        int ans = 1, x = 0;
        for(int i = 1; i < n; i++) {
            if(s[i] != s[i - 1]) ans++;
            if(i == n - 1) break;
            
            if(s[i] != s[i - 1] && s[i] != s[i + 1]) {
                if(s[i + 1] == s[i - 1]) x = 2;
                else x = max(x, 1);
            }
        }
        
        cout << ans - x << endl;
    }
}